pandas/core/common.py

```diff
@@ -1482,10 +1482,11 @@ def _get_dtype_from_object(dtype):
 
         try:
             return _get_dtype_from_object(getattr(np,dtype))
```

Comment:
On Thu, Jan 7, 2016, 1:09 PM Jeff Reback <notifications@github.com> wrote:

> In pandas/core/common.py
> <https://github.com/pydata/pandas/pull/11991#discussion_r49112021>:
>
> > @@ -1482,10 +1482,11 @@ def _get_dtype_from_object(dtype):
> >
> >          try:
> >              return _get_dtype_from_object(getattr(np,dtype))
>
> hmm, I think this should be changed to:
>
> np.dtype(dtype) as that wontt have the issue, and catch TypeError and not
> AttributeError
>
> and is more correct
>

I tried that. There was a test for dtype='integer' that fails. If you think
it shouldn't be supported, I would agree. But I don't feel strongly about
it.

>